lib/iex/lib/iex/autocomplete.ex

```diff
@@ -336,8 +372,42 @@ defmodule IEx.Autocomplete do
     format_hint(name, hint)
   end
 
+  defp to_hint(%{kind: :map_key, name: name, value_is_map: true}, hint) when name == hint do
+    format_hint(name, hint) <> "."
+  end
+
+  defp to_hint(%{kind: :map_key, name: name}, hint) do
+    format_hint(name, hint)
+  end
+
   defp format_hint(name, hint) do
     hint_size = byte_size(hint)
     :binary.part(name, hint_size, byte_size(name) - hint_size)
   end
+
+  defp get_evaluator do
+    case IEx.Server.local do
+      nil -> nil
+      pid ->
+        {:dictionary, dictionary} = Process.info(pid, :dictionary)
+        dictionary[:evaluator]
+    end
+  end
+
+  defp get_variable_value(ast_node, evaluator) do
+    binding = IEx.Evaluator.get_binding(evaluator)
+
+    {_, value} = Macro.postwalk(ast_node, {:binding, binding}, fn
```

Comment:
It's there to walk the AST when you are dealing with nested map access like here:

https://github.com/elixir-lang/elixir/pull/5488/files/ef2d240154d4fdb2b6c23cbd0fa2cf84af1bcaa4#diff-9f4689bfff76a705f5caf00680a347b6R134

The binding is just the starting accumulator.

Is there a better way to walk the nested map access AST to get the value out of it?